frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx

```diff
@@ -73,17 +74,37 @@ export async function action({ request }: ActionFunctionArgs) {
   const formData = Object.fromEntries(await request.formData());
   const parsedDataResult = await formDataSchema.safeParseAsync(formData);
 
+  const wsaddressWebService = await getWSAddressService();
+
   if (!parsedDataResult.success) {
     return json({
       errors: parsedDataResult.error.flatten(),
       formData: formData as Partial<z.infer<typeof formDataSchema>>,
     });
   }
+  const validateResultAddress = await wsaddressWebService.validateWSAddress({
+    addressLine: parsedDataResult.data.address,
+    city: parsedDataResult.data.city,
+    country: parsedDataResult.data.country,
+    language: 'EN',
+    postalCode: parsedDataResult.data.postalCode ?? '',
+    province: parsedDataResult.data.province ?? '',
+    geographicScope: 'GS',
+    parseType: 'PT',
+  });
+
+  console.debug('VALIDATE RETURN ' + JSON.stringify(validateResultAddress, null, 2));
 
   const sessionService = await getSessionService();
   const session = await sessionService.getSession(request);
   session.set('newHomeAddress', parsedDataResult.data);
-
+  if (validateResultAddress.statusCode === 'Invalid') {
+    return redirect('/personal-information/home-address/address-accuracy', {
+      headers: {
+        'Set-Cookie': await sessionService.commitSession(session),
+      },
+    });
+  }
   return redirect('/personal-information/home-address/confirm', {
```

Comment:
**Nitpick:** We can do something like this so we don't have multiple `return redirect(..)`:

```typescript
const redirectUrl = validateResultAddress.statusCode === 'Invalid' ? '/personal-information/home-address/address-accuracy' : '/personal-information/home-address/confirm';
return redirect(redirectUrl, {
  headers: {
    'Set-Cookie': await sessionService.commitSession(session),
  },
});
```

Reply:
done